src/network.rs

```diff
@@ -316,7 +360,16 @@ pub fn serve(port: u16) {
                                 data: encoder.compress_vec(data).unwrap(),
                             };
                             let encoded_msg = serialize(&msg, Infinite).unwrap();
-                            sockfd.send_to(&encoded_msg, &addr).unwrap();
+                            let mut encrypted_msg = encoded_msg.clone();
+							let tag_len = aead::CHACHA20_POLY1305.tag_len();
+							for _ in 0..tag_len {
```

Comment:
How about this: `encrypted_req_msg .resize(encoded_req_msg.len() + tag_len, 0)`?